Building on OS X Fails

Since commit cc2b57bc building on OS X fails. Here is the error output:

```
[ 24%] Built target elektra-ease
[ 24%] Linking CXX shared library ../../../../lib/libelektratools.dylib
Undefined symbols for architecture x86_64:
  "_elektraPluginClose", referenced from:
      kdb::tools::Plugin::uninit() in plugin.cpp.o
  "_elektraPluginOpen", referenced from:
      kdb::tools::Plugin::Plugin(kdb::tools::PluginSpec const&, kdb::KeySet&) in plugin.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [lib/libelektratools.dylib] Error 1
make[1]: *** [src/libs/tools/src/CMakeFiles/elektratools.dir/all] Error 2
make: *** [all] Error 2
```

If you need any additional information, then please just comment below.

> Can you shuffle the line touched in 6bd83ae around until it works?

Yeah, I will try it out. The following code

```rb
%w(elektratools elektra-core elektra-plugin elektra-ease).permutation.map { |permutation| puts permutation.join(' ') }
```
shows 24 permutations
```
elektratools elektra-core elektra-plugin elektra-ease
elektratools elektra-core elektra-ease elektra-plugin
elektratools elektra-plugin elektra-core elektra-ease
elektratools elektra-plugin elektra-ease elektra-core
elektratools elektra-ease elektra-core elektra-plugin
elektratools elektra-ease elektra-plugin elektra-core
elektra-core elektratools elektra-plugin elektra-ease
elektra-core elektratools elektra-ease elektra-plugin
elektra-core elektra-plugin elektratools elektra-ease
elektra-core elektra-plugin elektra-ease elektratools
elektra-core elektra-ease elektratools elektra-plugin
elektra-core elektra-ease elektra-plugin elektratools
elektra-plugin elektratools elektra-core elektra-ease
elektra-plugin elektratools elektra-ease elektra-core
elektra-plugin elektra-core elektratools elektra-ease
elektra-plugin elektra-core elektra-ease elektratools
elektra-plugin elektra-ease elektratools elektra-core
elektra-plugin elektra-ease elektra-core elektratools
elektra-ease elektratools elektra-core elektra-plugin
elektra-ease elektratools elektra-plugin elektra-core
elektra-ease elektra-core elektratools elektra-plugin
elektra-ease elektra-core elektra-plugin elektratools
elektra-ease elektra-plugin elektratools elektra-core
elektra-ease elektra-plugin elektra-core elektratools
```
Should I really try all of them?

> (or use -v to see invocation might also help, maybe -lelektra-plugin is missing after all)

The command `make VERBOSE=1` shows the following, which does not seem that helpful:

```
[ 24%] Linking CXX shared library ../../../../lib/libelektratools.dylib
cd "/Users/rene/Dropbox/Studium/Master Thesis/Configuration Parsing/Source/libelektra/build/src/libs/tools/src" && /usr/local/Cellar/cmake/3.4.3/bin/cmake -E cmake_link_script CMakeFiles/elektratools.dir/link.txt --verbose=1
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++   -std=c++11  -Wno-deprecated-declarations  -Wno-missing-field-initializers -Wold-style-cast -Woverloaded-virtual  -Wsign-promo  -Wno-long-long -pedantic -Wno-variadic-macros -Wall -Wextra -Wno-overlength-strings -Wsign-compare -Wfloat-equal -Wformat-security -Wshadow -Wcomments -Wtrigraphs -Wundef -Wuninitialized -Winit-self -O2 -g -DNDEBUG -dynamiclib -Wl,-headerpad_max_install_names  -o ../../../../lib/libelektratools.dylib -install_name @rpath/libelektratools.dylib CMakeFiles/elektratools.dir/backend.cpp.o CMakeFiles/elektratools.dir/backendbuilder.cpp.o CMakeFiles/elektratools.dir/backendparser.cpp.o CMakeFiles/elektratools.dir/backends.cpp.o CMakeFiles/elektratools.dir/helper/comparison.cpp.o CMakeFiles/elektratools.dir/helper/keyhelper.cpp.o CMakeFiles/elektratools.dir/merging/automergeconfiguration.cpp.o CMakeFiles/elektratools.dir/merging/automergestrategy.cpp.o CMakeFiles/elektratools.dir/merging/importmergeconfiguration.cpp.o CMakeFiles/elektratools.dir/merging/interactivemergestrategy.cpp.o CMakeFiles/elektratools.dir/merging/mergeconflictstrategy.cpp.o CMakeFiles/elektratools.dir/merging/mergeresult.cpp.o CMakeFiles/elektratools.dir/merging/metamergestrategy.cpp.o CMakeFiles/elektratools.dir/merging/newkeystrategy.cpp.o CMakeFiles/elektratools.dir/merging/onesidemergeconfiguration.cpp.o CMakeFiles/elektratools.dir/merging/onesidestrategy.cpp.o CMakeFiles/elektratools.dir/merging/onesidevaluestrategy.cpp.o CMakeFiles/elektratools.dir/merging/overwritemergeconfiguration.cpp.o CMakeFiles/elektratools.dir/merging/threewaymerge.cpp.o CMakeFiles/elektratools.dir/modules.cpp.o CMakeFiles/elektratools.dir/plugin.cpp.o CMakeFiles/elektratools.dir/plugindatabase.cpp.o CMakeFiles/elektratools.dir/plugins.cpp.o CMakeFiles/elektratools.dir/pluginspec.cpp.o CMakeFiles/elektratools.dir/specreader.cpp.o ../../../../lib/libelektra-plugin.0.8.14.dylib ../../../../lib/libelektra-ease.0.8.14.dylib ../../../../lib/libelektra-core.0.8.14.dylib -Wl,-rpath,"/Users/rene/Dropbox/Studium/Master Thesis/Configuration Parsing/Source/libelektra/build/lib"
Undefined symbols for architecture x86_64:
  "_elektraPluginClose", referenced from:
      kdb::tools::Plugin::uninit() in plugin.cpp.o
  "_elektraPluginOpen", referenced from:
      kdb::tools::Plugin::Plugin(kdb::tools::PluginSpec const&, kdb::KeySet&) in plugin.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [lib/libelektratools.dylib] Error 1
make[1]: *** [src/libs/tools/src/CMakeFiles/elektratools.dir/all] Error 2
make: *** [all] Error 2
```

I see a very similar problem when cross-compiling (OpenWRT):
```
/home/lambda/olinuxino/openwrt/staging_dir/toolchain-arm_arm926ej-s_gcc-5.2.0_musl-1.1.11_eabi/bin/arm-openwrt-linux-muslgnueabi-g++   -Os -pipe -march=armv5te
-mtune=arm926ej-s -fno-caller-saves -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -mfloat-abi=soft   -std=c++11  -Wno-deprecated-declarations  -Wstrict-null-sentinel -D_GLIBCXX_USE_NANOSLEEP -Wno-missing-field-initializers -Wold-style-cast -Woverloaded-virtual  -Wsign-promo  -Wno-long-long
-pedantic -Wno-variadic-macros -Wall -Wextra -Wno-overlength-strings -Wsign-compare -Wfloat-equal -Wformat-security -Wshadow -Wcomments -Wtrigraphs -Wundef -Wuninitialized -Winit-self -Wmaybe-uninitialized -DNDEBUG  -L/home/lambda/olinuxino/openwrt/staging_dir/target-arm_arm926ej-s_musl-1.1.11_eabi/usr/lib -L/home/lambda/olinuxino/openwrt/staging_dir/target-arm_arm926ej-s_musl-1.1.11_eabi/lib -L/home/lambda/olinuxino/openwrt/staging_dir/toolchain-arm_arm926ej-s_gcc-5.2.0_musl-1.1.11_eabi/usr/lib -L/home/lambda/olinuxino/openwrt/staging_dir/toolchain-arm_arm926ej-s_gcc-5.2.0_musl-1.1.11_eabi/lib CMakeFiles/backend.dir/backend.cpp.o
-o ../../../../bin/backend -rdynamic ../../../../lib/libelektratools.so ../../../../lib/libelektra-plugin.so.0.8.14 ../../../../lib/libelektra-ease.so.0.8.14 ../../../../lib/libelektra-core.so.0.8.14 -ldl
../../../../lib/libelektratools.so: undefined reference to `elektraPluginClose'
../../../../lib/libelektratools.so: undefined reference to `elektraPluginOpen'
collect2: error: ld returned 1 exit status
src/libs/tools/examples/CMakeFiles/backend.dir/build.make:102: recipe for target 'bin/backend' failed
```

--- a/src/libs/tools/src/CMakeLists.txt
+++ b/src/libs/tools/src/CMakeLists.txt
@@ -18,7 +18,7 @@ if (BUILD_SHARED)
 		"ELEKTRA_SHARED"
 		)
 
-	target_link_libraries (elektratools elektra-core elektra-plugin elektra-ease)
+	target_link_libraries (elektratools elektra-core elektra-kdb elektra-plugin elektra-ease)
 
 	if (${LD_ACCEPTS_VERSION_SCRIPT})
 		set_target_properties (elektratools PROPERTIES

